src/modules/musig/tests_impl.h

```diff
@@ -440,34 +433,35 @@ int musig_state_machine_diff_signers_combine_nonce_test(secp256k1_pubkey *combin
     return secp256k1_musig_session_combine_nonces(ctx, &session, signers_to_use, 2, NULL, NULL);
 }
 
-/* Recreates a session with the given session_id, signers, pk, msg etc. parameters
- * and tries to sign and verify the other signers partial signature. Both should fail
- * if msg is NULL. */
-int musig_state_machine_missing_msg_test(secp256k1_pubkey *pks, secp256k1_pubkey *combined_pk, unsigned char *pk_hash, unsigned char *nonce_commitment_other, secp256k1_pubkey *nonce_other, secp256k1_musig_partial_signature *partial_sig_other, unsigned char *sk, unsigned char *session_id, unsigned char *msg) {
+/* Initializaes a session with the given session_id, signers, pk, msg etc.
+ * parameters but without a message. Will test that the message must be
+ * provided with `get_public_nonce`.
+ */
+void musig_state_machine_late_msg_test(secp256k1_pubkey *pks, secp256k1_pubkey *combined_pk, unsigned char *pk_hash, unsigned char *nonce_commitment_other, secp256k1_pubkey *nonce_other, unsigned char *sk, unsigned char *session_id, unsigned char *msg) {
     secp256k1_musig_session session;
     secp256k1_musig_session_signer_data signers[2];
     unsigned char nonce_commitment[32];
     const unsigned char *ncs[2];
     secp256k1_pubkey nonce;
     secp256k1_musig_partial_signature partial_sig;
-    int partial_sign, partial_verify;
 
-    CHECK(secp256k1_musig_session_initialize(ctx, &session, signers, nonce_commitment, session_id, msg, combined_pk, pk_hash, 2, 1, sk) == 1);
+    CHECK(secp256k1_musig_session_initialize(ctx, &session, signers, nonce_commitment, session_id, NULL, combined_pk, pk_hash, 2, 1, sk) == 1);
     ncs[0] = nonce_commitment_other;
     ncs[1] = nonce_commitment;
-    CHECK(secp256k1_musig_session_get_public_nonce(ctx, &session, signers, &nonce, ncs, 2) == 1);
+
+    /* Trying to get the nonce without providing a message fails. */
+    CHECK(secp256k1_musig_session_get_public_nonce(ctx, &session, signers, &nonce, ncs, 2, NULL) == 0);
```

Comment:
Should we `ARG_CHECK` this violation?

Reply:
If the rationale for `ARG_CHECK`ing is still "everything that the rust typesystem would have caught" then I guess no.

Reply:
We could prevent this with session types, i.e. have a `Session<NonceUnset>` and `Session<NonceSet>`

Reply:
I considered this but thought that this is not exactly the rust type system as you could do something similar with C. But actually this is what we'll be doing in rust. Fixing.

Reply:
Fixed